src/overlays/actors/ovl_Demo_Du/z_demo_du.c

```diff
@@ -1,228 +1,1083 @@
+    s32 didAnimFinished;
+
+    DemoDu_UpdateBgCheckInfo(this, globalCtx);
+    didAnimFinished = DemoDu_UpdateSkelAnime(this);
+    DemoDu_CsFireMedallion_AdvanceTo05(this, didAnimFinished);
+}
+
+void DemoDu_UpdateCs_FM_05(DemoDu* this, GlobalContext* globalCtx) {
+    DemoDu_UpdateBgCheckInfo(this, globalCtx);
+    DemoDu_UpdateSkelAnime(this);
+    DemoDu_CsFireMedallion_AdvanceTo06(this, globalCtx);
+}
+
+void DemoDu_UpdateCs_FM_06(DemoDu* this, GlobalContext* globalCtx) {
+    DemoDu_UpdateBgCheckInfo(this, globalCtx);
+    DemoDu_UpdateSkelAnime(this);
+}
+
+void DemoDu_InitCs_GoronsRuby(DemoDu* this, GlobalContext* globalCtx) {
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_06011CA8, NULL, NULL, NULL, 0);
+    this->updateIndex = CS_GORONSRUBY_SUBSCENE(0);
+}
+
+// Cutscene: Darunia gives Link the Goron's Ruby.
+// Sfx played when Darunia lands at the floor at the start of the cutscene.
+void DemoDu_CsPlaySfx_GoronLanding(DemoDu* this) {
+    func_80078914(&this->actor.projectedPos, NA_SE_EN_GOLON_LAND_BIG);
+}
+
+// Cutscene: Darunia gives Link the Goron's Ruby.
+// Sfx played when Darunia is falling at the start of the cutscene.
+void DemoDu_CsPlaySfx_DaruniaFalling(GlobalContext* globalCtx) {
+    if (globalCtx->csCtx.frames == 160) {
+        func_800788CC(NA_SE_EV_OBJECT_FALL);
+    }
+}
+
+// Cutscene: Darunia gives Link the Goron's Ruby.
+// .
+void DemoDu_CsPlaySfx_DaruniaHitsLink(GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    Vec3f* projectedPos = &player->actor.projectedPos;
+
+    func_80078914(projectedPos, NA_SE_EN_DARUNIA_HIT_LINK);
+    Audio_PlaySoundGeneral(NA_SE_VO_LI_DAMAGE_S_KID, projectedPos, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+}
+
+// Cutscene: Darunia gives Link the Goron's Ruby.
+void DemoDu_CsPlaySfx_HitBreast(DemoDu* this) {
+    func_80078914(&this->actor.projectedPos, NA_SE_EN_DARUNIA_HIT_BREAST - SFX_FLAG);
+}
+
+// Cutscene: Darunia gives Link the Goron's Ruby.
+// Sfx played when Link is escaping from the gorons at the end of the scene.
+void DemoDu_CsPlaySfx_LinkEscapeFromGorons(GlobalContext* globalCtx) {
+    Player* player;
+
+    if (globalCtx->csCtx.frames == 1400) {
+        player = PLAYER;
+        Audio_PlaySoundGeneral(NA_SE_VO_LI_FALL_L_KID, &player->actor.projectedPos, 4, &D_801333E0, &D_801333E0,
+                               &D_801333E8);
+    }
+}
+
+// Cutscene: Darunia gives Link the Goron's Ruby.
+// Sfx played when Link is surprised by Darunia falling from the sky.
+void DemoDu_CsPlaySfx_LinkSurprised(GlobalContext* globalCtx) {
+    Player* player;
+
+    if (globalCtx->csCtx.frames == 174) {
+        player = PLAYER;
+        Audio_PlaySoundGeneral(NA_SE_VO_LI_SURPRISE_KID, &player->actor.projectedPos, 4U, &D_801333E0, &D_801333E0,
+                               &D_801333E8);
+    }
+}
+
+void DemoDu_CsGoronsRuby_UpdateFaceTextures(DemoDu* this, GlobalContext* globalCtx) {
+    u16* frames = &globalCtx->csCtx.frames;
```

Comment:
a bit of an odd temp. You sure you can't just use `globalCtx->csCtx.frames` in each comparison?